src/routes/+page.svelte

```diff
@@ -0,0 +1,241 @@
+<script lang="ts">
+	import * as d3 from 'd3';
+	import type { DateEntry } from '$lib/types';
+
+	export let data;
+	let entries = data?.entries;
+
+	$: timerange = all;
+
+	const lastWeek = new Date(new Date().getTime() - 7 * 24 * 60 * 60 * 1000);
+	const lastMonth = new Date(new Date().getTime() - 7 * 4 * 24 * 60 * 60 * 1000);
+	const lastSixMonths = new Date(new Date().getTime() - 7 * 4 * 24 * 60 * 60 * 1000 * 6);
+	const lastYear = new Date(new Date().getTime() - 365 * 24 * 60 * 60 * 1000);
+	const all = new Date('2022-12-31');
+
+	$: category = null;
+
+	const gap = 0;
+	const padding = 32;
+	$: innerWidth = 1300;
+	$: innerHeight = 600;
+	$: width = innerWidth / 1.05;
+	$: height = innerHeight / 1.75;
+
+	let tidy = entries
+		.map((d: DateEntry) => [
+			{
+				date: d.date,
+				status: 'Images published with alternative text',
+				value: d.images_published_with_alt_text
+			},
+			{
+				date: d.date,
+				status: 'Images published',
+				value: d.images_published
+			}
+		])
+		.flat();
+
+	$: tidy = entries
+		.map((d: DateEntry) => [
+			{
+				date: d.date,
+				status: 'Images published with alternative text',
+				value: category
+					? JSON.parse(d.category_data)[category]?.images_published_with_alt_text ?? 0
+					: d.images_published_with_alt_text
+			},
+			{
+				date: d.date,
+				status: 'Images published',
+				value: category
+					? JSON.parse(d.category_data)[category]?.images_published ?? 0
+					: d.images_published
+			}
+		])
+		.flat()
+		.filter((entry) => entry.date > timerange.toISOString().split('T')[0]);
+
+	$: index = d3.rollup(
+		tidy,
+		(v) => Object.fromEntries(v.map((o) => [o.status, o.value])),
+		(d) => d.date
+	);
+
+	$: x = d3.scaleUtc(d3.extent(tidy.map((d) => new Date(d.date))) as Iterable<Number>, [
+		padding,
+		width - padding
+	]);
+	$: y = d3.scaleLinear(d3.extent(tidy.map((d) => d.value)) as Iterable<Number>, [
+		height - padding,
+		padding
+	]);
+
+	let gx: SVGGElement;
+	let gy: SVGGElement;
+	$: d3.select(gx).call(d3.axisBottom(x));
+	$: d3.select(gy).call(d3.axisLeft(y));
+
+	function handleMouseOver(d: MouseEvent, data: { date: string; values: Record<string, number> }) {
+		d3.select('.tooltip')
+			.style('display', 'block')
+			.style('left', d.pageX + 'px')
+			.style('top', d.pageY - 200 + 'px')
```

Comment:
I think `d3.pointer` (<https://d3js.org/d3-selection/events#pointer>) would be helpful here in getting the position of the tooltip closer to the mouse. By default, this will probably place the tooltip near the bottom right of the mouse. You'll probably want to move it up a little bit to be top right, or even top left of the mouse.

If you do use this, you probably will also have to set `position: relative;` on your figure.

Reply:
Hmm, okay. Right now with d3.pointer when the tooltip shows up it modifies the height of the screen, since it now has relative position and goes from display: none to display: block. Maybe an opacity change would suffice instead.

Another concern (the reason why I subtract 200 from the top) is that when the tooltip stays too close to the cursor, sometimes when moving the cursor to the right the graph "jitters" between showing the tooltip and not because the cursor is entering the tooltip instead of the next bar. 

So the solution for now is pushing it up... 🤔

Reply:
> Right now with d3.pointer when the tooltip shows up it modifies the height of the screen, since it now has relative position 

Hmm did you add relative position to the tooltip? I was suggesting adding relative positioning to the `figure` element. Apologies if that was not clear.

As for the other concern, the way to deal with it is usually to set `pointer-events: none` on the tooltip element.

It’s also fine to leave this as it is!